Weather/Weather/Controllers/NewCityController.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  NewCityController.swift
+//  Weather
+//
+//  Created by Oleg Yankiwskyi on 5/16/18.
+//  Copyright © 2018 Oleg Yankiwskyi. All rights reserved.
+//
+
+import UIKit
+
+class NewCityController: UIViewController {
+
+    @IBOutlet weak var cityTextField: UITextField!
+    let citiesModel = CitiesModel()
+    
+    @IBAction func edit(_ sender: Any) {
+        self.cityTextField.backgroundColor = .clear
+    }
+    
+    @IBAction func tapButton(_ sender: Any) {
+        guard let city = cityTextField.text else {
+            self.cityTextField.backgroundColor = .red
+            return
+        }
+        citiesModel.isValidCity(city: city, complete: { isValid, cityName in
+            DispatchQueue.main.async {
+                if isValid {
+                    self.cityTextField.backgroundColor = .green
```

Comment:
Nobody will see your green bkg as soon as you immediately dismiss controller.
Split verification into two steps